nixos/modules/virtualisation/qcow.nix

```diff
@@ -0,0 +1,66 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+
+let
+  cfg = config.virtualisation.qcow;
+in
+{
+  options = {
+    virtualisation.qcow = {
+      baseImageSize = mkOption {
+        type = with types; either (enum [ "auto" ]) ints.positive;
+        default = "auto";
+        example = 8192;
+        description = ''
+          The base size of the QCOW2 image in MiB.
+        '';
+      };
+      additionalImageSpace = mkOption {
+        type = types.int;
+        default = 0;
+        example = 512;
+        description = ''
+          Additional disk space in MiB to be added to the image if <literal>virtualisation.qcow.baseImageSize</literal> is set to <literal>auto</literal>.
+        '';
+      };
+      imageBootPartitionSize = mkOption {
+        type = types.int.positive;
+        default = 256;
+        example = 512;
+        description = ''
+          The size of the boot partition, only takes effect if <literal>virtualisation.qcow.partitionTableType</literal> is set to <literal>efi</literal> or <literalhybrid</literal>.
+        '';
+      };
+      imagePartitionTableType = mkOption {
+        type = types.str;
+        default = "legacy";
+        example = "efi";
+        description = ''
+          Type of partition table to use for the image, available options are <literal>legacy</literal>, <literal>efi</literal>, <literal>legacy+gpt</literal>, <literal>hybrid</literal> or <literal>none</literal>. For additional information about eacch option: <link>https://github.com/NixOS/nixpkgs/blob/master/nixos/lib/make-disk-image.nix#L32</link>.
+        '';
+      };
+      filesystemType = mkOption {
+        type = types.str;
```

Comment:
maybe we want also an enum here? but not completely sure

Reply:
well after reading the code at https://github.com/NixOS/nixpkgs/blob/master/nixos/lib/make-disk-image.nix again. I guess we can remove this option, since only ext4 is supported anyway. and its ignored if partitonTableType is set to none

Reply:
Well, what if we make it possible to modify the filesystem too in nixos-generators?

Reply:
we would have to fix make-disk-image.nix first (or reimplement it). so I would do that in a new PR

Reply:
Done.